### PR TITLE
Type annotate code

### PR DESCRIPTION
**Pull request by cthoyt**

### CODE REVIEW

File: pybiopax/api.py [hunk 1]
```diff
@@ -91,37 +98,40 @@ def model_from_pc_query(kind, source, target=None, **query_params):
 
     Returns
     -------
-    pybiopax.biopax.BioPaxModel
+    :
         A BioPAX Model obtained from the results of the Pathway Commons query.
     """
     owl_str = graph_query(kind, source, target=target, **query_params)
+    if owl_str is None:
+        # TODO should this just return None or is an error appropriate?
```
Comment (cthoyt) [inline pybiopax/api.py:106]: @bgyori check this behavior

File: pybiopax/biopax/base.py [hunk 2]
```diff
@@ -100,17 +120,17 @@ def _simple_to_xml(self, attr, val):
...
 
     def __init__(self,
-                 standard_name=None,
-                 display_name=None,
-                 all_names=None,
-                 participant_of=None,
-                 availability=None,
-                 data_source=None,
-                 evidence=None,
+                 standard_name: Optional[str] = None,
```
Comment (cthoyt) [inline pybiopax/biopax/base.py:135]: these aren't right

File: pybiopax/biopax/base.py [hunk 3]
```diff
@@ -121,28 +141,53 @@ def __init__(self,
...
+        )
+
 
 class Gene(Entity):
     """BioPAX Gene"""
     def __init__(self, organism, **kwargs):
         super().__init__(**kwargs)
         self.organism = organism
 
+    def __repr__(self) -> str:
```
Comment (cthoyt) [inline pybiopax/biopax/base.py:167]: probably  remove this (and other custom reprs) before merge

File: pybiopax/biopax/interaction.py [hunk 4]
```diff
@@ -76,15 +77,15 @@ def __init__(self,
...
+    list_types: ClassVar[List[str]] = Interaction.list_types + \
         ['left', 'right', 'participant_stoichiometry']
 
     def __init__(self,
-                 left=None,
-                 right=None,
-                 conversion_direction=None,
-                 participant_stoichiometry=None,
-                 spontaneous=None,
+                 left: Optional[List] = None,
```
Comment (cthoyt) [inline pybiopax/biopax/interaction.py]: @bgyori list of what? BioPaxObject?

File: pybiopax/api.py (hunk 1 above)
Comment (bgyori) [reply pybiopax/api.py:106]: Interesting, I think it depends if owl_str being None indicates an empty result or if it indicates a technical error in the query call. If the latter, raising the error here or in the `graph_query` function makes sense. For the empty result case, returning None is better.

File: pybiopax/biopax/base.py (hunk 3 above)
Comment (bgyori) [reply pybiopax/biopax/base.py:167]: I think the repr I implemented in a higher-level class that just provides the class name and the standard name could remain, but I agree these detailed reprs could be reverted.

File: pybiopax/biopax/interaction.py (hunk 4 above)
Comment (bgyori) [reply pybiopax/biopax/interaction.py]: list of PhysicalEntity

File: pybiopax/api.py (hunk 1 above)
Comment (cthoyt) [reply pybiopax/api.py:106]: I think an empty result should still be a BioPaxModel, but just with an empty objects dictionary.

File: pybiopax/biopax/interaction.py (hunk 4 above)
Comment (cthoyt) [reply pybiopax/biopax/interaction.py]: ecd6a00

**Pull request closed without merging**